esprima.js

```diff
@@ -1274,27 +1281,13 @@
 
                     default:
                         if (isOctalDigit(ch)) {
```

Comment:
https://bugs.ecmascript.org/show_bug.cgi?id=3477 seems to suggest otherwise, that the behavior is web breaking. Not sure what to do here. @ariya?

I'm matching what I saw when I tested v8, that is:

```
`\08`
```

parses fine.